src/validator.cpp

```diff
@@ -180,7 +205,7 @@ uint32_t convertTextToUint32(const std::string &text, size_t initialValue = 0)
     uint32_t value = 0;
     size_t index = 0;
     for (size_t j = initialValue; j < 4; ++j) {
-        auto tempValue = static_cast<uint32_t>(text[index++]) << bitShifts[j];
+        auto tempValue = static_cast<uint32_t>((unsigned char)text[index++]) << bitShifts[j];
```

Comment:
I did have a reason for the double casting, wish you had asked me this when I made the change.

Reply:
Apparently, better late than never, so here we are. :)

Reply:
[1291e1f](https://github.com/cellml/libcellml/pull/1128/commits/1291e1f19e6b7020f2b7aacc026247d6f16c0ac0)

Reply:
It is because the text could theoritically be UTF-16 or UTF-32 and you can't cast those to `uint32_t` without warnings, but we don't allow those encodings so we are able to safely make the cast from UTF-8 text.

Reply:
Fair enough. Would it be worth make a note of it in our code?